spec/unit/matrix-client.spec.ts

```diff
@@ -165,12 +205,28 @@ describe("MatrixClient", function() {
             }
             return Promise.resolve(next.data);
         }
-        // Jest doesn't let us have custom expectation errors, so if you're seeing this then
-        // you forgot to handle at least 1 pending request. Check your tests to ensure your
-        // number of expectations lines up with your number of requests made, and that those
-        // requests match your expectations.
-        expect(true).toBe(false);
-        return new Promise(() => {});
+
+        const receivedRequestQueryString = new URLSearchParams(
+            convertQueryDictToStringRecord(queryParams),
+        ).toString();
+        const receivedRequest = decorateStringWithAnsiColor(
+            `${method} ${prefix}${path}${receivedRequestQueryString}`,
+            AnsiColorCode.Red,
+        );
+        const expectedQueryString = new URLSearchParams(
+            convertQueryDictToStringRecord(next.expectQueryParams),
+        ).toString();
+        const expectedRequest = decorateStringWithAnsiColor(
+            `${next.method} ${next.prefix ?? ''}${next.path}${expectedQueryString}`,
+            AnsiColorCode.Green,
+        );
+        // If you're seeing this then you forgot to handle at least 1 pending request.
+        throw new Error(
+            `A pending request was not handled: ${receivedRequest} ` +
+            `(next request expected was ${expectedRequest})\n` +
+            `Check your tests to ensure your number of expectations lines up with your number of requests ` +
+            `made, and that those requests match your expectations.`,
+        );
     }
 
     function makeClient() {
```

Comment:
The `Typescript Strict Error Checker` CI job is failing (which is required), https://github.com/matrix-org/matrix-js-sdk/actions/runs/3588517236/jobs/6039998561

Since this file is a minefield, it has thrown a bunch of error. I've gone through and made sure the remaining examples are part of completely unrelated adjacent code.

One of the heavy hitters is the untyped `client` but that lets out a barrage of more errors if we try to simply correct it by adding a type. I'm going to defer all of those to the future.

Need an exception here.

Reply:
This has resolved itself now that the codebase was updated to be strict.